framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2722,6 +2723,21 @@ private AnnotatedDeclaredType getFunctionalInterfaceType(Tree lambdaTree,
                 AnnotatedExecutableType methodExe = result.second;
                 return (AnnotatedDeclaredType) methodExe.getReturnType();
 
+            case CONDITIONAL_EXPRESSION:
+                ConditionalExpressionTree conditionalExpressionTree = (ConditionalExpressionTree) parentTree;
+                final AnnotatedTypeMirror falseType = getAnnotatedType(conditionalExpressionTree.getFalseExpression());
+                final AnnotatedTypeMirror trueType = getAnnotatedType(conditionalExpressionTree.getTrueExpression());
+                assertFunctionalInterface(javacTypes, (Type) falseType.getUnderlyingType(), parentTree, lambdaTree);
+                assertFunctionalInterface(javacTypes, (Type) trueType.getUnderlyingType(), parentTree, lambdaTree);
+                if (!falseType.equals(trueType)) {
```

Comment:
Do the two types really have to be equal? Isn't it enough that they are both subtypes of the overall type?